### PR TITLE
Add `slowTraceSegmentThreshold` to forcibly sample slow traces

### PR DESCRIPTION
**Pull request by zifeihan**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.
`slowTraceSegmentThreshold` is for you to open force save some slow trace segment when sampling mechanism active.When sampling mechanism activated, this config would make the slow trace segment sampled, ignoring the sampling rate.

### CODE REVIEW

File: oap-server/analyzer/agent-analyzer/src/test/java/org/apache/skywalking/oap/server/analyzer/provider/trace/TraceLatencyThresholdsAndWatcherTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        TraceLatencyThresholdsAndWatcher watcher = new TraceLatencyThresholdsAndWatcher(provider);
+        register.registerConfigChangeWatcher(watcher);
+        register.start();
+
+        while (watcher.getSlowTraceSegmentThreshold() == 10000) {
+            Thread.sleep(2000);
+        }
+        assertThat(watcher.getSlowTraceSegmentThreshold(), is(1000));
+        assertThat(provider.getModuleConfig().getSlowTraceSegmentThreshold(), is(500));
```
Comment (kezhenxu94) [inline oap-server/analyzer/agent-analyzer/src/test/java/org/apache/skywalking/oap/server/analyzer/provider/trace/TraceLatencyThresholdsAndWatcherTest.java:64]: why `is(500)`? Shouldn't it be `1000`?

**Review by kezhenxu94 — COMMENTED** [review]
LGTM generally, if you want to include this in the 8.2.0, please add a change log, as what you did in your previous PR

**Review by wu-sheng — COMMENTED** [review]
1. Please set the threshold of slow to at least 2000ms. Because in the traditional ToB systems, latency over 1s is very common.
1. This PR should be held until the 8.2.0 release.

File: oap-server/analyzer/agent-analyzer/src/test/java/org/apache/skywalking/oap/server/analyzer/provider/trace/TraceLatencyThresholdsAndWatcherTest.java (hunk 1 above)
Comment (zifeihan) [reply oap-server/analyzer/agent-analyzer/src/test/java/org/apache/skywalking/oap/server/analyzer/provider/trace/TraceLatencyThresholdsAndWatcherTest.java:64]: hello, `provider.getModuleConfig().getSlowTraceSegmentThreshold()` is module config value, watcher can't change it. watcher has it own value.

**Comment by zifeihan** [comment]
> 1. Please set the threshold of slow to at least 2000ms. Because in the traditional ToB systems, latency over 1s is very common.
> 2. This PR should be held until the 8.2.0 release.

done.

**Comment by kezhenxu94** [comment]
@zifeihan 8.2.0 is released, please fix the conflict

**Comment by zifeihan** [comment]
> @zifeihan 8.2.0 is released, please fix the conflict

done.

File: docs/en/setup/backend/configuration-vocabulary.md [hunk 2]
```diff
@@ -145,6 +145,7 @@ core|default|role|Option values, `Mixed/Receiver/Aggregator`. **Receiver** mode
 | - | - |forceSampleErrorSegment|When sampling mechanism activated, this config would make the error status segment sampled, ignoring the sampling rate.|SW_FORCE_SAMPLE_ERROR_SEGMENT|true|
 | - | - |segmentStatusAnalysisStrategy|Determine the final segment status from the status of spans. Available values are `FROM_SPAN_STATUS` , `FROM_ENTRY_SPAN` and `FROM_FIRST_SPAN`. `FROM_SPAN_STATUS` represents the segment status would be error if any span is in error status. `FROM_ENTRY_SPAN` means the segment status would be determined by the status of entry spans only. `FROM_FIRST_SPAN` means the segment status would be determined by the status of the first span only.|SW_SEGMENT_STATUS_ANALYSIS_STRATEGY|FROM_SPAN_STATUS|
 | - | - |noUpstreamRealAddressAgents|Exit spans with the component in the list would not generate the client-side instance relation metrics. As some tracing plugins can't collect the real peer ip address, such as Nginx-LUA and Envoy. |SW_NO_UPSTREAM_REAL_ADDRESS|6000,9000|
+| - | - |slowTraceSegmentThreshold|The threshold used to check the slow trace segment. Unit, millisecond.|SW_SLOW_TRACE_SEGMENT_THRESHOLD|2000|
```
Comment (kezhenxu94) [inline docs/en/setup/backend/configuration-vocabulary.md:148]:
```suggestion
| - | - |slowTraceSegmentThreshold|The threshold is used to check the slow trace segment. Unit, millisecond.|SW_SLOW_TRACE_SEGMENT_THRESHOLD|2000|
```

File: docs/en/setup/backend/dynamic-config.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ Right now, SkyWalking supports following dynamic configurations.
 |core.default.apdexThreshold| The apdex threshold settings, will override `service-apdex-threshold.yml`. | same as [`service-apdex-threshold.yml`](apdex-threshold.md) |
 |core.default.endpoint-name-grouping| The endpoint name grouping setting, will override `endpoint-name-grouping.yml`. | same as [`endpoint-name-grouping.yml`](endpoint-grouping-rules.md) |
 |agent-analyzer.default.sampleRate| Trace sampling , override `receiver-trace/default/sampleRate` of `applciation.yml`. | 10000 |
+|agent-analyzer.default.slowTraceSegmentThreshold| The threshold used to check the slow trace segment, override `receiver-trace/default/slowTraceSegmentThreshold` of `applciation.yml`. | 2000 |
```
Comment (kezhenxu94) [inline docs/en/setup/backend/dynamic-config.md:15]:
```suggestion
|agent-analyzer.default.slowTraceSegmentThreshold| The threshold is used to check the slow trace segment, override `receiver-trace/default/slowTraceSegmentThreshold` of `application.yml`. | 2000 |
```

please also fix the typo at line 14 👆

File: oap-server/server-bootstrap/src/main/resources/application.yml [hunk 4]
```diff
@@ -191,6 +191,7 @@ agent-analyzer:
     # Nginx and Envoy agents can't get the real remote address.
     # Exit spans with the component in the list would not generate the client-side instance relation metrics.
     noUpstreamRealAddressAgents: ${SW_NO_UPSTREAM_REAL_ADDRESS:6000,9000}
+    slowTraceSegmentThreshold: ${SW_SLOW_TRACE_SEGMENT_THRESHOLD:2000} # The threshold used to check the slow trace segment. Unit, millisecond.
```
Comment (kezhenxu94) [inline oap-server/server-bootstrap/src/main/resources/application.yml:194]:
```suggestion
    slowTraceSegmentThreshold: ${SW_SLOW_TRACE_SEGMENT_THRESHOLD:2000} # The threshold is used to check the slow trace segment. Unit, millisecond.
```

**Review by kezhenxu94 — COMMENTED** [review]
LGTM basically, just some nits, please take a look @zifeihan , thanks.

File: oap-server/server-bootstrap/src/main/resources/application.yml (hunk 4 above)
Comment (zifeihan) [reply oap-server/server-bootstrap/src/main/resources/application.yml]: thanks for your review, done.

**Review by kezhenxu94 — APPROVED** [review]
LGTM

File: docs/en/setup/backend/configuration-vocabulary.md [hunk 5]
```diff
@@ -145,6 +145,7 @@ core|default|role|Option values, `Mixed/Receiver/Aggregator`. **Receiver** mode
 | - | - |forceSampleErrorSegment|When sampling mechanism activated, this config would make the error status segment sampled, ignoring the sampling rate.|SW_FORCE_SAMPLE_ERROR_SEGMENT|true|
 | - | - |segmentStatusAnalysisStrategy|Determine the final segment status from the status of spans. Available values are `FROM_SPAN_STATUS` , `FROM_ENTRY_SPAN` and `FROM_FIRST_SPAN`. `FROM_SPAN_STATUS` represents the segment status would be error if any span is in error status. `FROM_ENTRY_SPAN` means the segment status would be determined by the status of entry spans only. `FROM_FIRST_SPAN` means the segment status would be determined by the status of the first span only.|SW_SEGMENT_STATUS_ANALYSIS_STRATEGY|FROM_SPAN_STATUS|
 | - | - |noUpstreamRealAddressAgents|Exit spans with the component in the list would not generate the client-side instance relation metrics. As some tracing plugins can't collect the real peer ip address, such as Nginx-LUA and Envoy. |SW_NO_UPSTREAM_REAL_ADDRESS|6000,9000|
+| - | - |slowTraceSegmentThreshold|The threshold is used to check the slow trace segment. Unit, millisecond.|SW_SLOW_TRACE_SEGMENT_THRESHOLD|2000|
```
Comment (wu-sheng) [inline docs/en/setup/backend/configuration-vocabulary.md:148]: The description is incorrect. What do you mean check slow trace?

Comment (wu-sheng) [reply docs/en/setup/backend/configuration-vocabulary.md:148]: This kind of description exists in many places.

**Review by wu-sheng — CHANGES_REQUESTED** [review]
Please make the document more accurate.

**Review by wu-sheng — COMMENTED** [review]
Miss changelog update?

**Comment by zifeihan** [comment]
> Miss changelog update?

done.

File: CHANGES.md [hunk 6]
```diff
@@ -12,6 +12,7 @@ Release Notes.
 
 #### OAP-Backend
 * Add the `@SuperDataset` annotation for BrowserErrorLog.
+* Support the slowly segment in the sampling mechanism.
```
Comment (wu-sheng) [inline CHANGES.md:15]:
```suggestion
* Support keeping collecting the slowly segments in the sampling mechanism.
```

**Review by wu-sheng — APPROVED** [review]

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**